src/viam/sdk/resource/resource_manager.cpp

```diff
@@ -1,58 +1,156 @@
 #include <viam/sdk/resource/resource_manager.hpp>
 
-#include <stdexcept>
+#include <exception>
+#include <memory>
 #include <string>
 #include <unordered_map>
 #include <vector>
 
-#include <viam/sdk/components/component_base.hpp>
-#include <viam/sdk/resource/resource_type.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/log/trivial.hpp>
+#include <boost/optional/optional.hpp>
+#include <grpcpp/impl/service_type.h>
+#include <grpcpp/support/status.h>
+
+#include <viam/api/component/generic/v1/generic.grpc.pb.h>
+
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/resource/resource.hpp>
+#include <viam/sdk/resource/resource_base.hpp>
+#include <viam/sdk/rpc/server.hpp>
 #include <viam/sdk/services/service_base.hpp>
 
 namespace viam {
 namespace sdk {
 
-ResourceManager::ResourceManager() {}
+std::shared_ptr<ResourceBase> ResourceManager::resource(std::string name) {
+    std::lock_guard<std::mutex> lock(lock_);
+
+    if (resources_.find(name) != resources_.end()) {
+        return resources_.at(name);
+    }
 
-// Register a new resource with the registry.
-// Resources may not have the same name.
-ResourceManager::ResourceManager(std::vector<std::shared_ptr<ResourceBase>> resources) {
-    for (auto& resource : resources) {
-        register_resource(resource);
+    if (short_names_.find(name) != short_names_.end()) {
+        std::string short_name = short_names_.at(name);
+        if (resources_.find(short_name) != resources_.end()) {
+            return resources_.at(short_name);
+        }
     }
+    return nullptr;
```

Comment:
[nit] This seems like we should throw an exception here. Otherwise we have a lot of code that directly dereferences something that could be nullptr.